### ISSUE TITLE
Disable map dragging

### ISSUE THREAD

**Issue by Mathieu-R** [alex3165/react-mapbox-gl#543]
Hello,

Is there any way to disable map dragging ?
I would like to draw areas on the map so I should disable dragging feature in that moment.

Thanks !

**Comment by Wykks** [comment]
Hi!

Just set `dragPan` map prop to `false` :)

**Comment by Mathieu-R** [comment]
I already tried and it didn't work :/

**Comment by Wykks** [comment]
Oops indeed, I thought that was in map props.

**Comment by Mathieu-R** [comment]
Found ! 😄 
I can disable drag with the `disable()` method from dragPan property.

```js 
addExclusionZone(evt) {
  // this.map is the "map ref"
  this.map.state.map.dragPan.disable();
}

**Comment by Wykks** [comment]
Yeah, you can always do it like that, but I think this should be in map props instead of factory in order to be able to do it the "react way".
I'll keep this open until it's done.

**Comment by daniel-hauser** [comment]
We could add a `dragPan` prop to `<Map>` but we should not remove `dragPan` from the Factory parameters.
